backend/GeometryParser.cpp

```diff
@@ -0,0 +1,128 @@
+#include "GeometryParser.h"
+#include <iostream>
+
+GeometryParser::GeometryParser () {
+    this->objectNames = {"point", "circle", "line", "triangle", "square", "rectangle"};
+    this->functionNames = {"angle", "intersection"};
+}
+
+std::string GeometryParser::identify (std::string s) {
+    // check for object
+    for (int i = 0; i < this->objectNames.size(); ++i) {
+        std::size_t found = s.find(this->objectNames[i]);
+
+        if (found != std::string::npos) {
+            return this->objectNames[i];
+        }
+    }
+
+    // check for function
+    for (int i = 0; i < this->functionNames.size(); ++i) {
+        std::size_t found = s.find(this->functionNames[i]);
+
+        if (found != std::string::npos) {
+            return this->functionNames[i];
+        }
+    }
+
+    return "";
+}
+
+std::unordered_map <std::string, Object*> GeometryParser::parseObject (std::string s) {
+    std::string objName = "";
+    int objParams = 0;
+
+    // find the object
+    for (int i = 0; i < this->objectNames.size(); ++i) {
```

Comment:
Here, we also only accept lowercase user input.

Reply:
True, haven't thought about that. I will change it.